api/include/opentelemetry/logs/logger.h

```diff
@@ -35,578 +36,277 @@ namespace logs
-   * @param attributes The attributes of the log as an initializer list
-   */
-  void Debug(nostd::string_view message,
-             std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>>
-                 attributes) noexcept
-  {
-    this->Log(Severity::kDebug, message, attributes);
-  }
-
-  /** Info severity overloads **/
-
-  /**
-   * Writes a log with a severity of info.
-   * @param message The message to log
-   */
-  void Info(nostd::string_view message) noexcept { this->Log(Severity::kInfo, message); }
-
-  /**
-   * Writes a log with a severity of info.
-   * @param attributes The attributes of the log as a key/value object
-   */
-  template <class T,
-            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
-  void Info(const T &attributes) noexcept
-  {
-    this->Log(Severity::kInfo, attributes);
-  }
-
-  /**
-   * Writes a log with a severity of info.
-   * @param message The message of the log
-   * @param attributes The attributes of the log as a key/value object
-   */
-  template <class T,
-            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
-  void Info(nostd::string_view message, const T &attributes) noexcept
-  {
-    this->Log(Severity::kInfo, message, attributes);
-  }
-
-  /**
-   * Writes a log with a severity of info.
-   * @param attributes The attributes of the log as an initializer list
+   * @tparam args Arguments which can be used to set data of log record by type.
+   *  Severity                                -> severity, severity_text
+   *  string_view                             -> body
+   *  AttributeValue                          -> body
+   *  SpanContext                             -> span_id,tace_id and trace_flags
+   *  SpanId                                  -> span_id
+   *  TraceId                                 -> tace_id
+   *  TraceFlags                              -> trace_flags
+   *  SystemTimestamp                         -> timestamp
+   *  system_clock::time_point                -> timestamp
+   *  KeyValueIterable                        -> attributes
+   *  Key value iterable container            -> attributes
+   *  span<pair<string_view, AttributeValue>> -> attributes(return type of MakeAttributes)
    */
-  void Info(std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>>
-                attributes) noexcept
+  template <class... ArgumentType>
+  void Debug(ArgumentType &&... args) noexcept
   {
-    this->Log(Severity::kInfo, attributes);
+    this->EmitLogRecord(Severity::kDebug, std::forward<ArgumentType>(args)...);
   }
 
   /**
    * Writes a log with a severity of info.
-   * @param message The message of the log
-   * @param attributes The attributes of the log as an initializer list
+   * @tparam args Arguments which can be used to set data of log record by type.
+   *  Severity                                -> severity, severity_text
```

Comment:
Should we not mention Severity as argument here, and also in Warn(), Error() and Fatal() ?

Reply:
My mistake and thanks.I also add a type trait to prevent to pass `Severity` as args.